staging/src/k8s.io/client-go/util/flowcontrol/throttle.go

```diff
@@ -66,44 +87,72 @@ type realClock struct{}
 func (realClock) Now() time.Time {
 	return time.Now()
 }
+
 func (realClock) Sleep(d time.Duration) {
 	time.Sleep(d)
 }
 
+func (realClock) Since(ts time.Time) time.Duration {
+	return time.Since(ts)
+}
+
+var (
+	_ Clock              = (*realClock)(nil)
+	_ clock.PassiveClock = (*realClock)(nil)
+)
+
 // NewTokenBucketRateLimiterWithClock is identical to NewTokenBucketRateLimiter
 // but allows an injectable clock, for testing.
 func NewTokenBucketRateLimiterWithClock(qps float32, burst int, c Clock) RateLimiter {
 	limiter := rate.NewLimiter(rate.Limit(qps), burst)
-	return newTokenBucketRateLimiter(limiter, c, qps)
+	return newTokenBucketRateLimiterWithClock(limiter, c, qps)
 }
 
-func newTokenBucketRateLimiter(limiter *rate.Limiter, c Clock, qps float32) RateLimiter {
+// NewTokenBucketPassiveRateLimiterWithClock is similar to NewTokenBucketRateLimiterWithClock
+// except that it returns a PassiveRateLimiter which does not have Accept() and Wait() methods
+// and uses a PassiveClock.
+func NewTokenBucketPassiveRateLimiterWithClock(qps float32, burst int, c clock.PassiveClock) PassiveRateLimiter {
+	limiter := rate.NewLimiter(rate.Limit(qps), burst)
+	return newTokenBucketRateLimiterWithPassiveClock(limiter, c, qps)
+}
+
+func newTokenBucketRateLimiterWithClock(limiter *rate.Limiter, c Clock, qps float32) *tokenBucketRateLimiter {
 	return &tokenBucketRateLimiter{
-		limiter: limiter,
-		clock:   c,
-		qps:     qps,
+		clockIndependentRateLimiter: clockIndependentRateLimiter{limiter: limiter, qps: qps},
+		clock:                       c,
 	}
 }
 
-func (t *tokenBucketRateLimiter) TryAccept() bool {
-	return t.limiter.AllowN(t.clock.Now(), 1)
+func newTokenBucketRateLimiterWithPassiveClock(limiter *rate.Limiter, c clock.PassiveClock, qps float32) *tokenBucketPassiveRateLimiter {
+	return &tokenBucketPassiveRateLimiter{
+		clockIndependentRateLimiter: clockIndependentRateLimiter{limiter: limiter, qps: qps},
+		clock:                       c,
+	}
 }
 
-// Accept will block until a token becomes available
-func (t *tokenBucketRateLimiter) Accept() {
-	now := t.clock.Now()
-	t.clock.Sleep(t.limiter.ReserveN(now, 1).DelayFrom(now))
+func (cirl *clockIndependentRateLimiter) Stop() {
 }
 
-func (t *tokenBucketRateLimiter) Stop() {
+func (cirl *clockIndependentRateLimiter) QPS() float32 {
+	return cirl.qps
 }
 
-func (t *tokenBucketRateLimiter) QPS() float32 {
-	return t.qps
+func (cirl *clockIndependentRateLimiter) Wait(ctx context.Context) error {
```

Comment:
This makes `*tokenBucketPassiveRateLimiter` implement `Wait`, which you do not want.